zerver/migrations/0431_alter_and_migrate_realm_name_in_notifications.py

```diff
@@ -0,0 +1,95 @@
+# Generated by Django 4.1.5 on 2023-01-11 20:21
+
+from django.db import migrations, models
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+# We include a copy of this structure as it was at the time this
+# migration was merged, since future should not impact the migration.
+REALM_NAME_IN_NOTIFICATIONS_AUTOMATIC = 1
+REALM_NAME_IN_NOTIFICATIONS_YES = 2
+REALM_NAME_IN_NOTIFICATIONS_NO = 3
+
+
+# The value of 'realm_name_in_notifications' for those users
+# who have manually changed the value of 'realm_name_in_notifications' as 'true'
+# should be updated as 'Yes', not 'Automatic'
+def update_realm_name_in_notifications_new_values(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    userprofiles = UserProfile.objects.filter(realm_name_in_notifications=True)
+    userprofiles.update(realm_name_in_notifications_new=REALM_NAME_IN_NOTIFICATIONS_YES)
+
+
+def reverse_code(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    userprofiles = UserProfile.objects.filter(
+        realm_name_in_notifications_new=REALM_NAME_IN_NOTIFICATIONS_YES
+    )
+    userprofiles.update(realm_name_in_notifications=True)
```

Comment:
The intermediate variable should probably be removed.

Reply:
You mean -- avoiding the involvement of `realm_name_in_notifications_new` logic. right?

I have updated the PR with the approach suggested by you (using `_policy` suffix) in the next review comment -- the new approach removes the intermediate variable.

Reply:
No, I mean you can just write `UserProfile.objects.filter(realm_name_in_notifications=True).update(...`. 

As a side note, `userprofiles` without an underscore is variable naming style we use in the project -- words are always separated by `_`s`.

Reply:
Done.